jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -2159,12 +1706,200 @@ ecma_builtin_typedarray_prototype_includes (ecma_value_t this_arg, /**< this arg
+      return ecma_builtin_typedarray_prototype_filter (this_arg, &info, arguments_list_p[0], arguments_list_p[1]);
+
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_REVERSE:
+    {
+      return ecma_builtin_typedarray_prototype_reverse (this_arg, &info);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SET:
+    {
+      return ecma_builtin_typedarray_prototype_set (this_arg, arguments_list_p[0], arguments_list_p[1]);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SUBARRAY:
+    {
+      return ecma_builtin_typedarray_prototype_subarray (&info, arguments_list_p[0], arguments_list_p[1]);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FILL:
+    {
+      return ecma_builtin_typedarray_prototype_fill (this_arg,
+                                                     &info,
+                                                     arguments_list_p[0],
+                                                     arguments_list_p[1],
+                                                     arguments_list_p[2]);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SORT:
+    {
+      if (!ecma_is_value_undefined (arguments_list_p[0]) && !ecma_op_is_callable (arguments_list_p[0]))
+      {
+        return ecma_raise_type_error (ECMA_ERR_MSG ("Callback function is not callable."));
+      }
+
+      return ecma_builtin_typedarray_prototype_sort (this_arg, &info, arguments_list_p[0]);
+
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FIND:
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FIND_INDEX:
+    {
+      bool is_find = (builtin_routine_id == ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FIND) ? true : false;
+      return ecma_builtin_typedarray_prototype_find_helper (this_arg,
+                                                            &info,
+                                                            arguments_list_p[0],
+                                                            arguments_list_p[1],
+                                                            is_find);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_INDEX_OF:
+    {
+      return ecma_builtin_typedarray_prototype_index_of (&info, arguments_list_p, arguments_number);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_LAST_INDEX_OF:
+    {
+      return ecma_builtin_typedarray_prototype_last_index_of (&info, arguments_list_p, arguments_number);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_COPY_WITHIN:
+    {
+      return ecma_builtin_typedarray_prototype_copy_within (this_arg, &info, arguments_list_p, arguments_number);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SLICE:
+    {
+      return ecma_builtin_typedarray_prototype_slice (&info, arguments_list_p, arguments_number);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_TO_LOCALE_STRING:
+    {
+      return ecma_builtin_typedarray_prototype_to_locale_string (&info);
+    }
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_KEYS:
+    case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_ENTRIES:
+    {
+      static const uint8_t mode[] =
```

Comment:
No static array is needed.

```
iterator_id = builtin_id - ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_KEYS + ECMA_ITERATOR_KEYS;
```

Reply:
The suggestion you had does not work because
```typedef enum
{
  ECMA_ITERATOR_KEYS,     /**< keys iterator */
  ECMA_ITERATOR_VALUES,   /**< values iterator */
  ECMA_ITERATOR_ENTRIES,  /**< entries iterator */
  ECMA_ITERATOR__COUNT,   /**< number of iterator kinds */
} ecma_iterator_kind_t;
```

As you see `ECMA_ITERATOR_VALUES` is between them so I would still prefer to use the separate array but do correct me if I misunderstood something.